pkg/resolvconf/resolvconf.go

```diff
@@ -0,0 +1,323 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Package resolvconf provides utility code to query and update DNS configuration in /etc/resolv.conf
+// originally from https://github.com/moby/moby/blob/6014c1e29dc34dffa77fb5749cc3281c1b4854ac/libnetwork/resolvconf/resolvconf.go
+package resolvconf
```

Comment:
thanks,  if you have used functions from moby or other, please put the reference to the original function
example  https://github.com/containerd/nerdctl/blob/10a62bdcdca37206d4c406174d83db5770c3e0da/cmd/nerdctl/events.go#L57

Reply:
I copied the whole file. Should I put the comment on top of each function ?